pkgs/development/python-modules/shinychat/default.nix

```
@@ -0,0 +1,125 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+
+  # build-system
+  hatch-vcs,
+  hatchling,
+
+  # dependencies
+  htmltools,
+
+  # optional-dependencies
+  anthropic,
+  chatlas,
+  google-generativeai,
+  langchain-core,
+  ollama,
+  openai,
+  pydantic,
+  tokenizers,
+
+  # tests
+  pillow,
+  playwright,
+  pytest-playwright,
+  pytestCheckHook,
+  shiny,
+  shinychat,
+}:
+
+buildPythonPackage rec {
+  pname = "shinychat";
+  version = "0.2.8";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "posit-dev";
+    repo = "shinychat";
+    tag = "py/v${version}";
+    hash = "sha256-thdLaZ+rnD8yumxhjXOLhufcSBD0oNKOWSxxDdJ9tNU=";
+  };
+
+  build-system = [
+    hatch-vcs
+    hatchling
+  ];
+
+  pythonRemoveDeps = [
+    "shiny" # circular dependency
+  ];
+  dependencies = [
+    htmltools
+  ];
+
+  optional-dependencies = {
+    providers = [
+      anthropic
+      chatlas
+      google-generativeai
+      langchain-core
+      ollama
+      openai
+      pydantic
+      tokenizers
+    ];
+  };
+
+  passthru.tests.pytest = shinychat.overridePythonAttrs {
```

```suggestion
  passthru.tests.pytest = shinychat.overridePythonAttrs {
    doCheck = true;
  };
```
and include `nativeCheckInputs` in the normal derivation.